lua/SS13.lua

```diff
@@ -0,0 +1,79 @@
+local SS13 = {}
+
+SS13.SSlua = dm.global_vars:get_var("SSlua")
+
+SS13.global_proc = "some_magic_bullshit"
+
+local states = SS13.SSlua:get_var("states"):to_table()
+for _, state in pairs(states) do
+	if state:get_var("internal_id") == dm.state_id then
+		SS13.state = state
+		break
+	end
+end
+
+function SS13.new(type, ...)
+	local args = table.pack(...)
+	args["n"] = nil
+	local datum = dm.global_proc("_new", type, args)
+	local references = SS13.state:get_var("references")
+	references:add(datum)
+	return datum
+end
+
+function SS13.await(thing_to_call, proc_to_call, ...)
+	if thing_to_call == nil then
+		thing_to_call = SS13.global_proc
+	end
+	if thing_to_call == SS13.global_proc then
+		proc_to_call = "/proc/" .. proc_to_call
+	end
+	local args = table.pack(thing_to_call, proc_to_call, ...)
+	args["n"] = nil
+	local promise = SS13.new("/datum/promise", table.unpack(args))
```

Comment:
What is the point of this entire pack/unpack fiasco, especially if we're not using n? Why not just `{ thing_to_call, proc_to_call, ...}` in `SS13.new` directly?

Reply:
Didn't know that about Lua(u)